src/screen.c

```diff
@@ -2547,10 +2547,10 @@ text_to_screenline(win_T *wp, char_u *text, int col)
 	{
 #ifdef FEAT_RIGHTLEFT
 	    if (wp->w_p_rl)
-		STRNCPY(current_ScreenLine, text, len);
+		mch_memmove(current_ScreenLine, text, (size_t)len + 1);
```

Comment:
I think should not add `+ 1` to len because the size of `current_ScreenLine` is `Columns` so `+ 1` can cause an out of bounds write.

Reply:
Is cast needed?